gcc/testsuite/rust/compile/auto_traits2.rs

```diff
@@ -15,12 +15,11 @@ fn foo(a: &(dyn A + Send + Sync)) {
 struct S;
 
 impl A for S {
-    fn a_method(&self) {}
+    fn a_method(&self) {} // { dg-warning "unused name" }
 }
 
 fn main() {
     let s = S;
 
-    foo(&s); // { dg-error "bounds not satisfied" }
-             // { dg-error "mismatched type" "" { target *-*-* } .-1 }
```

Comment:
I'm unsure about what has changed here, care to give more details?

Reply:
so now that we can actually apply auto traits, this code will work because `S` is automatically `Send` and `Sync`